src/collections/componentSet.ts

```diff
@@ -612,42 +637,19 @@ export class ComponentSet extends LazyCollection<MetadataComponent> {
     }
     return destructiveChangesTypes;
   }
+}
 
-  /**
-   * Each {@link SourceComponent} counts as an element in the set, even if multiple
-   * ones map to the same `fullName` and `type` pair.
-   *
-   * @returns number of metadata components in the set
-   */
-  public get size(): number {
-    let size = 0;
-    for (const collection of this.components.values()) {
-      // just having an entry in the parent map counts as 1
-      size += collection.size === 0 ? 1 : collection.size;
-    }
-    return size;
-  }
-
-  public get destructiveChangesPre(): Map<string, Map<string, SourceComponent>> {
-    return this.destructiveComponents[DestructiveChangesType.PRE];
-  }
-
-  public get destructiveChangesPost(): Map<string, Map<string, SourceComponent>> {
-    return this.destructiveComponents[DestructiveChangesType.POST];
-  }
-
-  private sourceKey(component: SourceComponent): string {
-    const { fullName, type, xml, content } = component;
-    return `${type.name}${fullName}${xml ?? ''}${content ?? ''}`;
-  }
```

Comment:
So one downside of this pattern of standalone function over the private methods is all three of these look like they could benefit from unit tests but aren't reachable for testing. If these are private methods you can always unit test them with some TS workarounds. 
You could export them, but doing that just for testing isn't ideal.  

thoughts?

Reply:
I'd prefer to export locally (not top-level) for UT vs. forcing TS to let you violate encapsulation to test private methods.